main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -394,6 +394,26 @@ object TypeError {
     })
   }
 
+  /**
+   * Java method not found type error.
+   *
+   * @param tpe  the type of the method.
+   * @param renv the rigidity environment.
+   * @param loc  the location where the error occured.
+   */
+  case class MethodNotFound(methodName: String, tpe: Type, renv: RigidityEnv, loc: SourceLocation)(implicit flix: Flix) extends TypeError {
```

Comment:
In terms of arguments, I think we need:

methodName
receiver object type
argument object types

A list of possible candidate methods on receiverObject type

But we don't need RigidityEnv here.